lit/docs/operation/tracing.lit

```diff
@@ -8,21 +8,104 @@ Tracing in Concourse enables the delivery of traces related to the internal
+It currently integrates with \link{Jaeger}{https://www.jaegertracing.io/} and
+\link{Google Cloud Trace}{https://cloud.google.com/trace} (Stackdriver),
 although support for other systems is planned to expand as the underlying SDK
 (\link{OpenTelemetry}{https://opentelemetry.io/}) evolves.
 
 
 \section{
   \title{Configuring Tracing}
 
-  There's only one variable that is required to be set in order to leverage
-  Jaeger's integration with Concourse:
+  To export spans to Jaeger, specify the Thrift HTTP endpoint of the Jaeger
+  collector:
 
   \codeblock{bash}{{{
   CONCOURSE_TRACING_JAEGER_ENDPOINT=http://jaeger:14268/api/traces
   }}}
 
-  This tells Concourse how to target Jaeger's Thrift HTTP endpoint to send the
-  traces to.
+  To export spans to Google Cloud Trace, specify the GCP Project ID:
+
+  \codeblock{bash}{{{
+  CONCOURSE_TRACING_STACKDRIVER_PROJECTID=your-gcp-project-id
+  }}}
+
+  Note that suitable GCP credentials must be available, via the usual
+  \link{\code{GOOGLE_APPLICATION_CREDENTIALS} environment variable}{https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable},
+  the default location that the \code{gcloud} CLI expects, or from GCP's
+  metadata server (if Concourse is deployed on GCP).
+}
+
+\section{
+  \title{What's emitted?}
+
+  Below is a summary of the various operations that Concourse currently traces.
+  They are arranged like a call tree, so that for operation described, its
+  sub-operations are described indented immediately below.
+
+  \list{
+    \code{scanner.Run} -- One of the \reference{resource-checker-components}
+    responsible for determining which checks need to be run.
+    \list{
+      \code{scanner.check} -- This operation simply represents inserting the
+      check in the database.
+    }
+  }{
+    \code{checker.Run} -- follows from \code{scanner.check} above and will
+    appear under the same trace. This is where the \reference{checker} is
+    actually executing a \reference{resource-check} script.
+  }{
+    \code{scheduler.Run} -- This represents one tick of the \reference{scheduler}.
+    \list{
+      \code{schedule-job} -- this is the same operation scoped to a single job.
+        \list{
+        \code{Algorithm.Compute} -- this is where the \reference{algorithm}
+        determines inputs for a job. Each of the resolvers below describes a
+        different strategy for determining inputs, depending on the job's config.
+        \list{
+          \code{individualResolver.Resolve} -- This is used to determine versions
+          input to a \reference{get-step} without
+          \reference{schema.step.get-step.passed}{\code{passed}} constraints.
+        }{
+          \code{groupResolver.Resolve} -- This is the juicy part of the algorithm,
+          which deals with
+          \reference{schema.step.get-step.passed}{\code{passed}} constraints.
+          \list{
+            \code{groupResolver.tryResolve} \list{
```

Comment:
Not sure if there's anything you can do about this (it might just be a Booklit quirk) but the font-size on this nested list seems just slightly off (`1em` vs the usual `0.9em` that `code` in a `p` tag gets), making it look slightly wonky (but really not terrible):

<img width="506" alt="Screen Shot 2020-06-01 at 10 36 12 AM" src="https://user-images.githubusercontent.com/26583442/83419937-e1450a00-a3f3-11ea-95ec-48d5aabc22fd.png">

...whereas if we can somehow move this list into the `p` that holds the text above:

<img width="453" alt="Screen Shot 2020-06-01 at 10 39 05 AM" src="https://user-images.githubusercontent.com/26583442/83420110-25380f00-a3f4-11ea-8b05-81904adc633c.png">

...it looks a little better

Reply:
...this looks easier said than done. @vito pls? maybe it makes sense to drop the descriptions of all those private functions under groupResolver.

Reply:
Hmm this brings up an interesting point around sustainability with documenting traces. It feels like if we go too far they're likely to become out of date. Should we have a convention where public functions are documented but private ones aren't? That way I can more freely refactor the internal bits, while knowing that any changes to "public" spans will require a documentation update.

re: the formatting - the syntax here feels kind of odd because it starts as a paragraph but then leads to a `\list`, which is a block element that can't exist in a paragraph (i.e. you can't have a `<ul>` in a `<p>`).

I think if you break the `\list` onto its own line it might do what you want it to - Booklit should parse the `\code` line as its own paragraph, which I think will make the font size consistent, since I think our CSS specifies `p code { font-size: 0.9em }` or something.

Reply:
Ah good point - I didn't piece together that the lowercase spans were intended to be "private". I assume there's no distinction between "Public" and "private" operations from Jaeger's perspective, right?

Reply:
I might go a step further and suggest that private functions should not be traced, i.e. "if it's worth tracing, it's worth making part of the public interface"

Reply:
Out of curiosity, is there any parallel notion to "log level" for traces? I guess something similar could be achieved with "tags"

Reply:
the concept I've heard of is "sampling priority" - I've heard it mentioned in [opentracing best practices](https://opentracing.io/docs/best-practices/#setting-sampling-priority-before-the-trace-starts) and seen in [client libraries](https://github.com/opentracing/opentracing-go/blob/master/ext/tags.go#L56-L57), but it's not actually part of the opentelemetry spec, it's just a common convention. https://github.com/jaegertracing/jaeger/issues/365 has some related discussion.

Reply:
@jamieklassen Hmm yeah I think that's going too far. I think traces are really valuable for helping diagnose problems during development, and in this case the private traces were instrumental (heh) in finding the root cause of an issue in Hush House.

To me, tracing is something that should have a very low barrier to entry, even if means we under-document them. (I have differing thoughts on logging, though. IMO we should be more careful about what we log and try to make them more human-friendly, while preferring tracing/metrics for actual troubleshooting.)

Reply:
ok i'll drop the private stuff, with the exception of `scanner.check` which I believe is quite important